addons/mail/static/src/models/thread/thread.js

```diff
@@ -1374,7 +1374,7 @@ function factory(dependencies) {
             // Include all the messages that are known locally but the server
             // didn't take into account.
             return this.orderedMessages.reduce((total, message) => {
-                if (message.id <= countFromId) {
+                if (message.id <= countFromId || message.isCurrentPartnerAuthor) {
```

Comment:
What is the reason behind this change? This method is hard to maintain and prone to errors so I don't like diff on it.

Reply:
This is to bring this in line with how message_unread_counter is computed in the backed. Currently, the two don't agree and there is flickering between the two values because of this. I could also change the backend code instead.

Reply:
I would prefer if the backend one was changed then.
Current user or not should not matter for the count, if a message is marked unread it is should be unread, even if own message.